### ISSUE TITLE
Should we release created Anchors?

### ISSUE THREAD

**Issue by itanbp** [google-ar/arcore-android-sdk#262]
Hi,
I see in [this](https://stackoverflow.com/a/45987189/1869297) answer that the startAnchor is removed from the Session.
`
session.removeAnchors(Collections.singleton(startAnchor));
`

I assume the `session.removeAnchors()` and `session.addAnchor()` are deprected since ARCore 1.0?

My questions are - in the hello_ar_java sample app, there is the `ArrayList<Anchor> anchors` list. 
1. should we release the Anchor resource? maybe calling [anchor.detach()](https://developers.google.com/ar/reference/java/com/google/ar/core/Anchor.html#detach())?
2. when we should we release those resources? onDestroy()?
3. are there any other resources we should release? In hello_ar_java  I noticed the `    surfaceView.onPause()` and `session.pause()`, is this sufficient?

Thanks!

**Comment by inio** [comment]
There's non-trivial computational cost in maintaining a Trackable, so it's good to detach Anchors whenever you no longer need them, allowing the underlying Trackables to be released.

As for when: whenever you no longer need them.  The garbage collector will handle releasing them normally, but if you create temporary anchors, you should explicitly detach them as soon as they are no longer needed.

**Comment by inio** [comment]
Closing for inactivity
